provision/setup-docker

```diff
@@ -1,16 +1,34 @@
 #!/usr/bin/env bash
-set -x
+set -ex
```

Comment:
More flexible version of the old script

- environment variable `DOCKER_LARGE_DISK` is used to indicate where to put docker's large files (we do this on the annex and I have it this way locally)
- all other arguments are users that will be granted permission to use the docker socket